.circleci/config.yml

```diff
@@ -259,12 +266,87 @@ jobs:
-          when: always
+
+      - when:
+          condition:
+              or:
+                  - equal: [ all, << parameters.test-type >> ]
+          steps:
+              - run:
+                  name: test
+                  # Necessary to explicitly use bash, otherwise gradlew's status code won't be received and the job will hang.
+                  shell: bash.exe
+                  command: yarn test --reporter mocha-junit-reporter --reporter-option mochaFile=test-results/mocha/test-results.xml
+                  when: always
+
+      - when:
+          condition:
+            equal: [ sfge, << parameters.test-type >> ]
+          steps:
+              - run:
+                  name: test-sfge
+                  # Necessary to explicitly use bash, otherwise gradlew's status code won't be received and the job will hang.
+                  shell: bash.exe
+                  command: |
+                      TESTGLOB=$(circleci tests glob "sfge/src/test/**/*Test.java" | circleci tests split --split-by=timings)
+                      echo $TESTGLOB
+                      TESTARRAY=($TESTGLOB)
+                      TESTARG=""
+                      for element in "${TESTARRAY[@]}"
+                      do
+                        TESTARG="$TESTARG --tests `basename $element .java`"
+                      done
+                      echo $TESTARG
+                      yarn test-sfge $TESTARG
+                  when: always
+
+      - when:
+          condition:
+            equal: [ cli-messaging, << parameters.test-type >> ]
+          steps:
+            - run:
+                name: test-cli-messaging
+                # Necessary to explicitly use bash, otherwise gradlew's status code won't be received and the job will hang.
+                shell: bash.exe
+                command: |
+                      if [[ $CIRCLE_NODE_INDEX -gt 0 ]]
+                      then
+                        exit 0
+                      fi
+                      yarn test-cli-messaging 
+                when: always
+
+      - when:
+          condition:
+              equal: [ pmd-cataloger, << parameters.test-type >> ]
+          steps:
+              - run:
+                  name: test-pmd-cataloger
+                  # Necessary to explicitly use bash, otherwise gradlew's status code won't be received and the job will hang.
+                  shell: bash.exe
+                  command: |
+                      if [[ $CIRCLE_NODE_INDEX -gt 0 ]]
+                      then
+                        exit 0
+                      fi
+                      yarn test-pmd-cataloger
+                  when: always
+
+      - when:
+          condition:
+              equal: [ ts, << parameters.test-type >> ]
+          steps:
+              - run:
+                  name: test-ts
+                  # Necessary to explicitly use bash, otherwise gradlew's status code won't be received and the job will hang.
```

Comment:
This comment is technically inaccurate, since the typescript tests don't use gradlew.
Replace it with something like:
```
# Explicitly using bash, for simplicity of required shell script.
```